calico_kubernetes/calico_kubernetes.py

```diff
@@ -141,8 +158,119 @@ def _configure_interface(self):
         print('Finished configuring network interface')
         return ep
 
+    def _container_add(self, container_name, workload_id, pid, ip, interface,
+                       hostname, orchestrator_id):
+        """
+        Add a container (on this host) to Calico networking with the given IP.
+
+        :param container_name: The name or ID of the container.
+        :param ip: An IPAddress object with the desired IP to assign.
+        :param interface: The name of the interface in the container.
+        """
+        # Check if the container already exists. If it does, exit.
+        try:
+            _ = self._datastore_client.get_endpoint(
+                hostname=hostname,
+                orchestrator_id=orchestrator_id,
+                workload_id=workload_id
+            )
+        except KeyError:
+            # Calico doesn't know about this container.  Continue.
+            pass
+        else:
+            print("This container has already been configured with Calico Networking.")
+            sys.exit(1)
+
+        # Obtain information from Docker Client and validate container state
+        self._validate_container_state(container_name)
+
+        # Obtain ip pool from datastore client, validate pool, and assign
+        # ip address through IPAM Client
+        self._ipam_client.assign_address(None, ip)
+
+        # Obtain next hops from datastore client, validate next hop, and
+        # create Endpoint object
+        ep = self._datastore_client.create_endpoint(hostname, orchestrator_id,
+                                                    workload_id, ip)
+        self._datastore_client.set_endpoint(ep)
```

Comment:
I don't think you need this set_endpoint() here.

If you replace the update_endpoint below with a set_endpoint, we should be fine.  No need to do two database writes instead of one.